examples/real_function_space.py

```diff
@@ -115,26 +115,60 @@ def u_exact(x):
 
 # We can now assemble the matrix and vector
 
-A = dolfinx.fem.petsc.assemble_matrix_block(a)
+try:
+    A = dolfinx.fem.petsc.assemble_matrix_block(a)
+except AttributeError:
+    A = dolfinx.fem.petsc.assemble_matrix(a, kind="mpi")
 A.assemble()
-b = dolfinx.fem.petsc.assemble_vector_block(L, a, bcs=[])
+
+
+# On the main branch of DOLFINx, the `assemble_vector` function for blocked spaces has been rewritten to reflect how
+# it works for standard assembly and `nest` assembly. This means that lifting is applied manually.
+# In this case, with no Dirichlet BC, we could skip those steps.
+# However, for clarity we include them here.
+
+bcs = []
+main_assembly = False
+try:
+    b = dolfinx.fem.petsc.assemble_vector_block(L, a, bcs=bcs)
+except AttributeError:
+    main_assembly = True
+    b = dolfinx.fem.petsc.assemble_vector(L, kind="mpi")
+    bcs1 = dolfinx.fem.bcs_by_block(dolfinx.fem.extract_function_spaces(a, 1), bcs)  # type: ignore
+    dolfinx.fem.petsc.apply_lifting(b, a, bcs=bcs1)  # type: ignore
+    b.ghostUpdate(PETSc.InsertMode.ADD, PETSc.ScatterMode.REVERSE)  # type: ignore
+    bcs0 = dolfinx.fem.bcs_by_block(dolfinx.fem.extract_function_spaces(L), bcs)  # type: ignore
+    dolfinx.fem.petsc.set_bc(b, bcs0)
```

Comment:
I have not seen this function being used before. Perhaps we should again add a link to more documentation? I think it is potentially confusing for the reader (at least for me) why we only call this function for `bcs0` and not `bcs1`.